src/clp_ffi_py/modules/clp_four_byte_encoder.cpp

```diff
@@ -0,0 +1,103 @@
+#include <clp_ffi_py/Python.hpp>
+#include <clp_ffi_py/encoder/encoding_methods.hpp>
+
+// NOLINTBEGIN(*-avoid-c-arrays)
+// NOLINTBEGIN(cppcoreguidelines-avoid-non-const-global-variables)
+// NOLINTBEGIN(modernize-use-trailing-return-type)
+// NOLINTBEGIN(readability-identifier-naming)
+
+PyDoc_STRVAR(
+        encode_preamble_doc,
+        "encode_preamble(ref_timestamp, timestamp_format, timezone)\n"
+        "--\n\n"
+        "Creates the encoded CLP preamble for a stream of encoded log messages"
+        " using 4-byte encoding.\n"
```

Comment:
```suggestion
        " using the 4-byte encoding.\n"
```
The `the` here makes a subtle difference. "4-byte encoding" might mean we just use 4 bytes to store the preamble or it might mean we're referring to some well known 4-byte encoding. "the 4-byte encoding" indicates that it's a specific type of encoding, not necessarily that it uses only 4 bytes or that it's well known.